packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+      })
+  } 
+
+  /**
+   * Check if stacking is enabled for next reward cycle
+   * 
+   * @returns {Promise<boolean>} that resolves to a bool if the operation succeeds
+   */
+  async stackingEnabledNextCycle(): Promise<boolean> {
+    return (await this.getPoxInfo()).rejection_votes_left_required > 0;
+  }
+
+  /**
+   * Check if account has minimum require amount of Stacks for stacking
+   * 
+   * @returns {Promise<boolean>} that resolves to a bool if the operation succeeds
+   */
+  async hasMinimumRequiredStxAmount(): Promise<boolean> {
+    const balance: BN = await this.getAccountBalance();
+    // TODO pox info should use string type instead of number
+    const min: BN = new BN((await this.getPoxInfo()).min_amount_ustx.toString());
+    return balance.gte(min);
+  }
+
+  /**
+   * Check if account can lock stx
+   * 
+   * @param {CanLockStxOptions} options - a required lock STX options object 
+   * 
+   * @returns {Promise<StackingEligibility>} that resolves to a StackingEligibility object if the operation succeeds
+   */
+  async canLockStx({
+    poxAddress, 
+    cycles
+  }: CanLockStxOptions
+  ): Promise<StackingEligibility> {
+    const balancePromise: Promise<BN> = this.getAccountBalance();
+    const poxInfoPromise = this.getPoxInfo();
+
+    return Promise.all([balancePromise, poxInfoPromise])
+      .then(([balance, poxInfo]) => {
+        const { hashMode, data } = convertBTCAddress(poxAddress);
+        const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+        const hashbytes = bufferCV(data);
+        const poxAddressCV = tupleCV({
+          hashbytes,
+          version: hashModeBuffer,
+        });
+
+        const [contractAddress, contractName] = poxInfo.contract_id.split('.');
+
+        return callReadOnlyFunction({
+          network: this.network,
+          contractName,
+          contractAddress,
+          functionName: 'can-stack-stx',
+          senderAddress: this.address,
+          functionArgs: [
+              poxAddressCV,
+              uintCV(balance.toString()),
+              uintCV(poxInfo.reward_cycle_id),
+              uintCV(cycles.toString())
+            ],
+        });
+      })
+      .then((responseCV: ClarityValue) => {
+        if(responseCV.type === ClarityType.ResponseOk) {
+          return {
+            eligible: true
+          };
+        } else {
+          const errorCV = responseCV as ResponseErrorCV;
+          return {
+            eligible: false,
+            reason: StackingErrors[+cvToString(errorCV.value)]
+          };
+        }
+      })
+  }
+
+  /**
+   * Generate and broadcast a stacking transaction to lock STX
+   * 
+   * @param {LockStxOptions} options - a required lock STX options object 
+   * 
+   * @returns {Promise<string>} that resolves to a broadcasted txid if the operation succeeds
+   */
+  async lockStx({
+    amountMicroStx,
+    poxAddress,
+    cycles,
+    key,
+    burnBlockHeight,
```

Comment:
do I understand correctly that this will default to the next cycle if not specified?

Reply:
we'd likely also consider error checking - past or current height shouldn't be accepted.

On a different note: Because the only options are block numbers in the future, we could also think about this field as a delta to current block height (`startInXBlocks`). For instance, starting in 3 blocks would just require an integer of 3. With that, developers don't have to think about the block height

Reply:
It would not default to any value. I think giving developers more control by setting the exact block here is better than making it slightly easier to use but taking away ability to be exact.

Reply:
I can see the tradeoffs of not being exact. mainly, I'm assuming that devs would want to get the latest block height right before calling this method. That could be avoided with a default and a delta, but I don't feel strongly either way. your call